hail/src/main/scala/is/hail/expr/ir/GenericLines.scala

```diff
@@ -160,7 +160,15 @@ object GenericLines {
             // move scanned input from buf to lineData
             val n = bufPos - begin
             if (linePos + n > lineData.length) {
-              val newLineData = new Array[Byte]((linePos + n) * 2)
+              val copySize = linePos.toLong + n
+              if (copySize > Int.MaxValue)
```

Comment:
Should this be copySize >= Int.MaxValue. 

scala> val newLineData = new Array[Byte](Int.MaxValue)
java.lang.OutOfMemoryError: Requested array size exceeds VM limit

Reply:
Hmm. This is an implementation limit, not a language limit:
https://www.baeldung.com/java-arrays-max-size

The link above suggests Int.MaxValue - 8

Reply:
Pushed up a fix.

Reply:
thanks for the comment!